### PR TITLE
Split transaction creation

### PR DESCRIPTION
**Pull request by ConorStevenson**

### CODE REVIEW

File: moonwalking/blocks/base.py [hunk 1]
```diff
@@ -50,8 +50,21 @@ def validate_addr(self, addr: str):
...
 
-    @abstractmethod
     async def send_money(self, priv: str, addrs: List[Tuple[str, D]]) -> str:
+        tx = await self.build_tx(priv, addrs)
+        signed = self.sign_tx(priv, tx)
+        return await self.broadcast_tx(signed)
+
+    @abstractmethod
+    async def build_tx(self, priv: str, addrs: List[Tuple[str, D]]):
+        raise NotImplementedError
```
Comment (markharley) [inline moonwalking/blocks/base.py]: Don't think we need to `raise` and use `@abstractmethod`

File: moonwalking/blocks/eth_generic.py [hunk 2]
```diff
@@ -209,3 +154,45 @@ def get_method_signature(method_name):
         elif to_string:
             return decode_abi(['string'], HexBytes(result))[0].decode()
         return result
+
+    async def send_eth(self, priv, addrs):
+        tx = await EthereumGeneric.build_tx(self, priv, addrs)
```
Comment (markharley) [inline moonwalking/blocks/eth_generic.py]: I'm still not sure why we'd want this. Seems like having it is really just repeating the base implementation

File: moonwalking/main.py [hunk 3]
```diff
@@ -144,7 +151,8 @@ def validate_addr(self, addr):
         if is_valid(addr):
             return self.to_legacy_address(addr)
 
-    async def send_money(self, priv, addrs):
+    async def build_tx(self, priv: str, addrs: List[Tuple[str, D]]):
+        # Todo: Need to break this up.
```
Comment (markharley) [inline moonwalking/main.py]: `Todo` still needed?

File: moonwalking/main.py [hunk 4]
```diff
@@ -281,3 +279,20 @@ class Lendingblock(EthereumGeneric):
             raise GeneralError("not enough eth in buffer wallet")
 
         return addr, priv
+
+    async def build_tx(self, priv: str, addrs: List[Tuple[str, D]]):
```
Comment (markharley) [inline moonwalking/main.py]: Does this method ever get called now that we're fixing the base classes method in `EthereumGeneric`?

File: moonwalking/blocks/base.py (hunk 1 above)
Comment (ConorStevenson) [reply moonwalking/blocks/base.py]: Agreed, and ideally we'd have a docstring rather than `pass`...

File: moonwalking/main.py (hunk 3 above)
Comment (ConorStevenson) [reply moonwalking/main.py]: Yeah, this is BCH

Comment (ConorStevenson) [reply moonwalking/main.py]: Note ```    def sign_tx(self, priv, tx):
        return tx```

**Pull request merged**